lib/inputs/select-input.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+var _ = require('underscore');
+var BaseInput = require('./base-input');
+var errors = require('../errors');
+
+class SelectInput extends BaseInput {
+    constructor(id, params) {
+        super('select', id, params);
+    }
+
+    getInputDefaultValue() {
+        return this.params.items[0].value;
+    }
+
+    normalizeParams(params) {
+        params = Object.assign({}, params);
+
+        if (!params.items) {
+            throw errors.compileError('INPUT-REQ-PARAMS-MISSING', {
+                input_id: this.id,
+                missing_params: 'items'
+            });
+        }
+
+        if (!_.isArray(params.items)) {
+            throw errors.compileError('INPUT-INVALID-PARAM', {
+                input_id: this.id,
+                param: 'items',
+                param_value: params.items,
+                message: 'Item param must be an array'
+            });
+        }
+
+        params.items = params.items.map((item) => {
+            // if value is string, number or array expand to obj with value and label
+            if (_.isString(item) || _.isNumber(item) || _.isArray(item)) {
```

Comment:
I think I’d actually prefer to expanding objects which lack `value` or `label`. It’s more likely the user makes a mistake of forgetting these properties than that he/she wants to actually expand an object.